### PR TITLE
Update chain extension example to show argument passing

### PR DESCRIPTION
**Pull request by justinFrevert**
This changes the `rand-extension` Ink! example code to pass along an argument from the contract to the chain extension in the runtime. This PR addresses the first ToDo [here](https://github.com/paritytech/ink/issues/891), which is a repeat user question. I verified this works in a Substrate node on my machine.

### CODE REVIEW

**Comment by justinFrevert** [comment]
The failing CI step seems to be unrelated to this change and prevalent among other current PRs.

File: examples/rand-extension/lib.rs [hunk 1]
```diff
@@ -88,9 +88,9 @@ mod rand_extension {
 
         /// Update the value from the runtimes random source.
```
Comment (cmichi) [inline examples/rand-extension/lib.rs:89]: Could you update this comment here and describe the effect of `subject`?

**Comment by cmichi** [comment]
> The failing CI step seems to be unrelated to this change and prevalent among other current PRs.

It's fixed now, it was an issue with one of the GitHub tokens.

**Comment by cmichi** [comment]
@justinFrevert Could you also merge `master`? We think the contract sizes which were posted might be due to your PR being behind `master`.

**Comment by justinFrevert** [comment]
Thanks for grabbing it, @HCastano

File: examples/rand-extension/lib.rs [hunk 2]
```diff
@@ -86,11 +86,12 @@ mod rand_extension {
             Self::new(Default::default())
         }
 
-        /// Update the value from the runtimes random source.
+        /// Seed a random value by passing some known argument `subject` to the runtime's random
+        /// source. Then, update the current value with the new random value
```
Comment (cmichi) [inline examples/rand-extension/lib.rs:90]:
```suggestion
        /// Seed a random value by passing some known argument `subject` to the runtime's
        /// random source. Then, update the current `value` stored in this contract with the
        /// new random value.
```
```

**Review by cmichi — APPROVED** [review]
Thanks @justinFrevert!

**Comment by cmichi** [comment]
@justinFrevert Are you up for answering the StackExchange question as well?

**Comment by justinFrevert** [comment]
> @justinFrevert Are you up for answering the StackExchange question as well?

[I gave it a shot](https://stackoverflow.com/questions/68332726/how-do-you-pass-and-handle-arguments-to-a-chainextension). Let me know how that looks.

**Comment by cmichi** [comment]
Great, thx! I'll also incorporate your answer into our `ink-docs`.

**Pull request merged**